### DISCUSSION TITLE
Hits with documents containing all the query words

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#211] [Feedback & Feature Proposal]
### Motivation 

Currently, the search engine returns documents containing all the query words but also documents containing one of the query words.
The users might want to disable this behavior and only get documents containing all the query words only.
See https://github.com/meilisearch/MeiliSearch/issues/1694.

### Solution

A solution could be to add a `optionalWords` settings (or another name), set to `true` by default. The default behavior is to return documents containing all the query words but also documents containing one of the query words.
If set to `false` only the documents containing all the words of the query are returned.

### Technical

This setting is already implemented on Milli's side, it just has to be exposed on MeiliSearch's side. The implementation of this feature should be quick.

**Comment by gmourier** [comment]
Thanks, @curquiza and @bnfd 😇

In the meantime, it is possible to have this behavior by wrapping each term with a [phrase search](https://docs.meilisearch.com/reference/features/search_parameters.html#phrase-search) expression.

e.g. `"Hotel" "Champvert"` will only return documents having the 2 terms `Hotel` and `Champvert`.

The limitation of the phrase search is that typographical errors are not taken into account.

**Comment by gmourier** [comment]
Locked in favor of #255
